src/Orleans/Messaging/BufferPool.cs

```diff
@@ -38,9 +39,9 @@ public string Name
             private set;
         }
 
-        internal static void InitGlobalBufferPool(IMessagingConfiguration config)
+        internal static void InitGlobalBufferPool(IOptions<MessagingOptions> messagingConfigurationOptions)
         {
-            GlobalPool = new BufferPool(config.BufferPoolBufferSize, config.BufferPoolMaxSize, config.BufferPoolPreallocationSize, "Global");
+            GlobalPool = new BufferPool(messagingConfigurationOptions.Value.BufferPoolBufferSize, messagingConfigurationOptions.Value.BufferPoolMaxSize, messagingConfigurationOptions.Value.BufferPoolPreallocationSize, "Global");
         }
```

Comment:
since this is setting the pattern, please access .Value just once first, and then use the options from it